sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/EventHubScope.cs

```diff
@@ -74,7 +85,10 @@ public async ValueTask DisposeAsync()
 
             try
             {
-                await ResourceManager.CreateRetryPolicy().ExecuteAsync(() => client.EventHubs.DeleteAsync(resourceGroup, eventHubNamespace, EventHubName));
+                if (WasEventHubCreated)
```

Comment:
nit:  No need to spend the resources/time on lines 83 and 84 when there is nothing to tear down.   Maybe add a block around line 80 for:

```csharp
if (!WasEventHubCreated)
{
    _disposed = true;
}
```

and remove the check on 88?    Not a blocker, we can tune this after the fact.